tests/app/main/test_frameworks.py

```diff
@@ -4124,3 +4125,211 @@ def test_error_if_box_not_ticked(self, data_api_client):
         assert len(
             doc.xpath('//span[@class="validation-message"][contains(text(), "You can only save and continue if you agree to the proposed changes")]')  # noqa
         ) == 1
+
+
+@mock.patch('app.main.views.frameworks.data_api_client', autospec=True)
+class TestReuseFrameworkSupplierDeclaration(BaseApplicationTest):
+    """Tests for frameworks/<framework_slug>/declaration/reuse view."""
+
+    def setup_method(self, method):
+        super(TestReuseFrameworkSupplierDeclaration, self).setup_method(method)
+        with self.app.test_client():
+            self.login()
+
+    def test_reusable_declaration_framework_slug_param(self, data_api_client):
+        """Ensure that when using the param to specify declaration we collect the correct declaration."""
+
+        # Modify the data client.
+        t07 = '2009-12-03T01:01:01.000000Z'
+        framework = {
+            'x_field': 'foo',
+            'allow_declaration_reuse': True,
+            'application_close_date': t07,
+            'slug': 'g-cloud-8',
+            'name': 'g-cloud-8'
+        }
+        declaration = {'status': 'complete'}
+
+        data_api_client.get_framework.return_value = {'frameworks': framework}
+        data_api_client.get_supplier_declaration.return_value = {'declaration': declaration}
+
+        # Do the get.
+        with self.client as cont:
+            resp = cont.get(
+                '/suppliers/frameworks/g-cloud-9/declaration/reuse?reusable_declaration_framework_slug=g-cloud-8'
+            )
+
+            # Assert request arg inside context manager.
+            assert request.args['reusable_declaration_framework_slug'] == 'g-cloud-8'
+
+        # Assert response OK.
+        assert resp.status_code == 200
+
+        # Assert expected api calls.
+        data_api_client.get_supplier_declaration.assert_called_once_with(1234, 'g-cloud-8')
+
+    def test_404_when_specified_declaration_not_found(self, data_api_client):
+        """Fail on a 404 if declaration is specified but not found."""
+
+        # Modify the data client.
+        framework = {}
+        data_api_client.get_framework.return_value = {'frameworks': framework}
+        data_api_client.get_supplier_declaration.side_effect = APIError()
```

Comment:
This could be `APIError(mock.Mock(status_code=404))` as I think 404 is what the API should return if it's working and doesn't find a declaration for that framework.  Raw `APIError()` on its own gives a 503.